src/map/itemdb.cpp

```diff
@@ -2997,11 +2996,46 @@ uint8 ItemGroupDatabase::pc_get_itemgroup(uint16 group_id, bool identify, map_se
 	}
 	if (group->random.empty())
 		return 0;
-	
-	// Get all the 'must' item(s) (subgroup 0)
+
+	size_t count = 0;
+	// Count all the 'must' item(s) (subgroup 0)
 	uint16 subgroup = 0;
 	std::shared_ptr<s_item_group_random> random = util::umap_find(group->random, subgroup);
 
+	if (random != nullptr && !random->data.empty()) {
+		count = random->data.size();
+	}
+
+	// Count all 'random' subgroups
+	for (const auto& random : group->random) {
+		if (random.first == 0 || random.second->data.empty())
+			continue;
+		count++;
+	}
+
+	// Check if the player has enough weight and space
+#ifdef RENEWAL
+	// Official servers use 10 as the minimum amount of slots required to get the items
+	// The <= is intentional, as in official servers you actually need an extra empty slot
+	if (pc_inventoryblank(sd) <= std::max<size_t>(count, 10) || pc_is70overweight(sd)) {
+		clif_msg_color(sd, MSI_PICKUP_FAILED_ITEMCREATE, color_table[COLOR_RED]);
```

Comment:
Here we only have one error message for both?
And even a different one?

Reply:
Yeah, they changed it for renewal, now it uses the line that says you're over 70% overweight or not having enough space